timestamp-impl/src/main/java/com/palantir/timestamp/AvailableTimestamps.java

```diff
@@ -65,25 +90,29 @@ public synchronized void fastForwardTo(long newMinimum) {
         upperLimit.increaseToAtLeast(newMinimum + ALLOCATION_BUFFER_SIZE);
     }
 
-    private long lastHandedOut() {
+    private synchronized long lastHandedOut() {
         return lastReturnedTimestamp.get();
     }
 
-    private synchronized TimestampRange handOutTimestamp(long targetTimestamp) {
-        checkArgument(
-                targetTimestamp > lastHandedOut(),
-                "Could not hand out timestamp '%s' as it was earlier than the last handed out timestamp: %s",
-                targetTimestamp, lastHandedOut());
-
-        TimestampRange rangeToHandOut = TimestampRange.createInclusiveRange(lastHandedOut() + 1, targetTimestamp);
+    private synchronized TimestampRange getRangeToHandOut(long targetTimestamp) {
+        long lastHandedOut = lastHandedOut();
+        if (targetTimestamp <= lastHandedOut) {
+            throw new IllegalArgumentException(String.format(
+                    "Could not hand out timestamp '%s' as it was earlier than the last handed out timestamp: %s",
+                    targetTimestamp, lastHandedOut));
+        }
+        return TimestampRange.createInclusiveRange(lastHandedOut + 1, targetTimestamp);
+    }
 
+    private synchronized TimestampRange handOutTimestamp(long targetTimestamp) {
+        TimestampRange rangeToHandOut = getRangeToHandOut(targetTimestamp);
         allocateEnoughTimestampsToHandOut(targetTimestamp);
         lastReturnedTimestamp.increaseToAtLeast(targetTimestamp);
 
         return rangeToHandOut;
     }
 
-    private void allocateEnoughTimestampsToHandOut(long timestamp) {
+    private synchronized void allocateEnoughTimestampsToHandOut(long timestamp) {
```

Comment:
note to reviewers: technically this was already safe, but want `synchronized` for documentation and avoids having to litter with `assert Thread.holdsLock(this);` that bloats bytecode more.